sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/expressions/ExpressionEvalHelper.scala

```diff
@@ -93,6 +95,40 @@ trait ExpressionEvalHelper extends GeneratorDrivenPropertyChecks {
     }
   }
 
+  protected def checkExceptionInExpression[T <: Throwable](
+      expression: => Expression,
+      inputRow: InternalRow,
+      expectedErrMsg: String)(
+      implicit classTag: ClassTag[T]): Unit = {
+    val clazz = classTag.runtimeClass
+    val serializer = new JavaSerializer(new SparkConf()).newInstance
+    val resolver = ResolveTimeZone(new SQLConf)
+    val expr = resolver.resolveTimeZones(
+      serializer.deserialize[Expression](serializer.serialize(expression)))
+
+    def checkException(eval: => Unit, testMode: String): Unit = {
```

Comment:
Can we use the existing `intercept[...]` infrastructure here (not sure if it is possible)?

Reply:
ok, I'll reconsider again. thanks.

Reply:
Thanks!